services/streammanager/streammanager.go

```diff
@@ -65,15 +69,14 @@ func Close(producer interface{}, destType string) error {
 }
 
 // Produce delegates call to appropriate manager based on parameter destination
-func Produce(jsonData json.RawMessage, destType string, producer, config interface{}) (int, string, string) {
+func Produce(jsonData json.RawMessage, destType string, producer common.StreamProducer, config map[string]interface{}) (int, string, string) {
 	switch destType {
 	case "KINESIS", "KAFKA", "AZURE_EVENT_HUB", "CONFLUENT_CLOUD", "PERSONALIZE",
 		"FIREHOSE", "EVENTBRIDGE", "GOOGLEPUBSUB", "GOOGLESHEETS", "BQSTREAM", "LAMBDA":
-		streamProducer, ok := producer.(common.StreamProducer)
-		if !ok {
-			return 400, "Could not create stream producer", "Could not create stream producer"
+		if producer == nil {
+			return 500, "producer is nil", "producer is nil"
```

Comment:
why are changing this to a retryable error?

Reply:
This is not user data error but rudder server went into a bad state so kept 500, should we panic in this case? what is the best way to handle it

Reply:
how did rudder server went into a bad state and what makes us believe that on next try things will be better?

Reply:
First thing first, we need to make sure this cannot happen in production because we cannot really recover from such a panic without altering the codebase to fix the underlying bug.
As long as we can ensure that, whether we return 500, 400 or panic is not really that much important to me :)
Now, excluding test cases and after going through the code I don't see how this can be nil in practice, so it should be ok to panic